File: src/main/java/com/core/base/corebase/repository/MemberRepository.java
package com.core.base.corebase.repository;

import com.core.base.corebase.domain.organization.Organization;
import com.core.base.corebase.domain.user.Member;
import com.core.base.corebase.domain.user.code.MemberState;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface MemberRepository extends JpaRepository<Member, Long> {
    Optional<Member> findByUser_UidAndOrganizationIdAndState(UUID uid, Long organizationId, MemberState state);
    List<Member> findByOrganization(Organization organization);
    Optional<Member> findByOrganizationAndId(Organization organization, Long id);
    boolean existsByEmailAndOrganization(String email, Organization organization);
    List<Member> findByEmailAndUserIsNull(String email);
}

File: src/main/kotlin/com/core/base/corebase/repository/MemberRepository.kt
package com.core.base.corebase.repository

import com.core.base.corebase.domain.organization.Organization
import com.core.base.corebase.domain.user.Member
import com.core.base.corebase.domain.user.code.MemberState
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface MemberRepository : JpaRepository<Member, Long> {
    fun findByUser_UidAndOrganizationIdAndState(uid: UUID, organizationId: Long, state: MemberState): Member?
    fun findByOrganization(organization: Organization): List<Member>
    fun findByOrganizationAndId(organization: Organization, id: Long): Optional<Member>
    fun existsByEmailAndOrganization(email: String, organization: Organization): Boolean
}